Add custom error text for filtered event results

sdotson opened the pull request:
This is for #124. Now that I think about it, it's possible I overthought this and what was asked for was a simple text change. Here it is in any case!

pfarnach commented:
@sdotson Your work looks good but I'm unclear on the goal of this ticket and I think we may either have to complicate this more or simplify it. 

@InciteDemocracy @pdw207 - can you clarify what we want to accomplish with this?  From the issue it seems like we want a message that fits what filter is set. Because results (or no results) from filtering are the totality of all of the filters, something like `No event in 10 miles of 01002` isn't totally accurate because you're talking about a specific date, and if you add in search text, you'd have to add that to include that in the message.  Thoughts?

InciteDemocracy commented:
The goal of this ticket as I understand it (and maybe @pdw207 has some additional thoughts) is to give people a special message when they set a distance filter that is too narrow and nothing comes up. I think it makes sense to nudge users toward setting a wider distance setting if they get no results on a search that includes a location with a range setting that is less than the max 500 miles. 

If there is no distance setting the message should just say: **No events**

If there is a distance filter set with a range of less than 500 miles, it should say something like **No events, (You might want to try a wider distance filter)**

pfarnach commented:
@InciteDemocracy Ok, so to jump into the details it sounds like we need four different messages:
1.) when no search text is set but there is a location/range -- e.g. "No events for selected date. Try setting a larger distance range"
2.) when search text is set with no location/range -- e.g. "No events for selected date. Try using different search terms"
3.) when both are set -- e.g. "No events for selected date. Try using different search terms or setting a larger distance range"
4.) when neither are set -- e.g. "No events for selected date"

If that sounds right, @sdotson you'll have to update the `hasFilters` bool value to an array or something (two bools?) to accommodate the different scenarios.

sdotson commented:
Sounds good. I'll pause on this until we get confirmation from @InciteDemocracy .

InciteDemocracy commented:
How about using “No events are coming up” which has the double meaning of upcoming in time, and also” coming up” from the search? This may make more sense than “No events for selected date” which implies that we only display events happening on that specific day. (we are planning to show events from that day forward)

It would go like this:

1.) when no search text is set but there is a location/range: “No events are coming up...Try setting a larger distance range”
2.) when search text is set with no location/range: “No events are coming up...Try searching for something else”
3.) when both are set: “No events are coming up...Try searching for something else or setting a larger distance range”
4.) when neither are set -- e.g. “No events are coming up”

sdotson commented:
Hello everybody. I updated the `EventsList` component to take in a prop of `filters` instead of `hasFilter`. I've also updated the conditional logic, text, and have added a spec file so we can test the `EventsList` component.

pfarnach commented on src/components/EventsList/EventsList.js:
```diff
@@ -12,19 +12,34 @@ function renderEvents(events) {
     />
   );
 
+  let noEventText = 'No events are coming up';
```
It seems like this logic should live inside the `if` statement below on line 25.  

A small thing but there should be spaces after each of the `...`s and before the `Try`s in the text.
